pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -1211,6 +1205,43 @@ IdealState ensureAllPartitionsConsuming(TableConfig tableConfig, PartitionLevelS
     return idealState;
   }
 
+  private void createNewConsumingSegment(TableConfig tableConfig, PartitionLevelStreamConfig streamConfig,
```

Comment:
Let's pass in the `startOffset` of the new segment, instead of `offsetFactory` and `isLatestSegmentOffline`

Reply:
Ack